python/paddle/v2/fluid/learning_rate_decay.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import layers
+from framework import Variable
+
+__all__ = ['exponential_decay', ]
+"""
+When training a model, it's often useful to decay the
+learning rate during training process, this is called
+learning_rate_decay. There are many strategies to do
+this, this module will provide some classic method.
+User can also implement their own learning_rate_decay
+strategy according to this module.
+"""
+
+
+def exponential_decay(learning_rate,
```

Comment:
I think we should implement a base class because both `exponential_decay` and `inverse_time_decay` have a lot of code in common. We can create a base class that contains most of the boilerplate code. The inherited classes, `exponential_decay` , `inverse_time_decay` and any other future learning rate decay schemes can only implement the mathematics for the decay/ What are your thoughts on this?

Reply:
the common code has been removed since we overload some Python Operator such as "+" "/" "*", it will convert the value in Python to Variable implicitly. So that codes are all removed